azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -392,6 +394,12 @@ private boolean canWriteInCurrentWorkingDirectory(final String effectiveUser)
     final List<String> checkIfUserCanWriteCommand = Arrays
         .asList(CREATE_FILE, getWorkingDirectory() + "/" + TEMP_FILE_NAME);
     final int result = executeAsUser.execute(effectiveUser, checkIfUserCanWriteCommand);
+    // If TEMP_FILE user_can_write is created, it should be deleted at the end of the function
+    try {
+      Files.deleteIfExists(Paths.get(getWorkingDirectory() + "/" + TEMP_FILE_NAME));
```

Comment:
Minor nit: save the filename instead of recomputing it twice:
```
final String tmpFilePath = getWorkingDirectory() + "/" + TEMP_FILE_NAME;
```
You might also consider using the logic within `Paths` to construct the hierarchical path, instead of string manipulation:
```
final Path tmpFilePath = Paths.get(getWorkingDirectory(), TEMP_FILE_NAME);
final List<String> checkIfUserCanWriteCommand = Arrays.asList(CREATE_FILE, tmpFilePath.toString());
...
```

Reply:
@xkrogen thank for the review. I have modified the code as suggested.